### DISCUSSION TITLE
what is the reason behind adopting hsl instead of hex or rgb?

### DISCUSSION THREAD

**Discussion by raiyansarker** [shadcn-ui/ui#1802] [General]
I understand for some people out there, the color can be understood easily given their values. But the way this ui library is setup, it is hard to see colors in the editor if you are not that comfortable with hsl. On top of that, I believe some libraries doesn't have hsl support. Recently I tried using shadcn UI with framer motion and it seems that framer motion doesn't support hsl css variables, or at least as much as I tried to solve the issue.

So why not adopt something like rgb?

**Comment by IRediTOTO** [comment]
I have the same question. I don't understand when look at this code...
![image](https://github.com/shadcn-ui/ui/assets/15695252/034e4d85-3af3-4d35-8d5d-dead1d62e09c)

**Reply by Th1nhNg0** [reply]
Each color is defined using the HSL (Hue, Saturation, Lightness) color model. The HSL values allow you to adjust the colors in terms of human color perception more intuitively than RGB.
For example, --primary: 276, 86%, 57%; means:
Hue: 276 (which is around a purple shade),
Saturation: 86% (a very vibrant version of the hue),
Lightness: 57% (neither too dark nor too light, a medium brightness).

**Reply by tauren** [reply]
The issue in my opinion is lack of editor tooling. Devs are accustomed to seeing their color palette within their editors, which can be a massive time saver.

In VSCode, color values in CSS files are rendered in color. But due to the way these color variables are used within this framework, `hsl()`, `rgb()`, etc should be excluded. The plain HSL and RGB values are not colorized in our editors. 

![image](https://github.com/shadcn-ui/ui/assets/282084/da92f0af-0c62-4bf0-bd91-d02f100c89d9)

Also note that fractional values are not colorized either. It might be best to round these numbers when they are generated.

Does anyone know of an editor plugin that can render these strings in color? It probably wouldn't be too hard to adapt an existing plugin to support it.

Edit: fixed the image

**Reply by tauren** [reply]
Found one!
https://github.com/shadcn-ui/ui/issues/215

Someone else had the same frustration and created a plugin:
https://marketplace.visualstudio.com/items?itemName=meouwu.css-var-color-decorator

**Reply by raiyansarker** [reply]
Looks good but I am not sure why it is not working in my project. 
<img width="454" alt="Screenshot 2024-01-29 at 7 53 42 PM" src="https://github.com/shadcn-ui/ui/assets/38852396/cc105c4d-f515-4288-b933-bd44fed54232">

**Reply by moroya** [reply]
> Looks good but I am not sure why it is not working in my project.

It is possible that the language setting is set to "PostCSS".
Adding "postcss" to "cssVarColorDecorator.supportedLanguages" made it work.

**Reply by CoolZonee** [reply]
I mean I don't need to understand the color by reading the hex or hsl or anything else, I just want easy to develop, the hsl just bring so much trouble...

**Reply by sleeyax** [reply]
Skill issue tbh. HSL is by far more 'easy to develop' than hex. Grasp the basics of HSL and you'll never want to use hex again...

**Comment by tauren** [comment]
> Looks good but I am not sure why it is not working in my project.

Yeah, I posted here too soon, before trying it out. I'm having the same problem you are:
https://github.com/shadcn-ui/ui/issues/215#issuecomment-1913107316

**Comment by yf-hk** [comment]
IMO it's not the best practice. At least let users choose to use HSL or RGB. I've never used HSL and all the design files, color pickers are using RGB which makes it so tidous to convert RGB to HSL. and, for what?

**Reply by sleeyax** [reply]
The reasoning behind it is already explained in the [theming section of the docs](https://ui.shadcn.com/docs/theming). It points to [this article](https://www.smashingmagazine.com/2021/07/hsl-colors-css/).

**Comment by mukuljainx** [comment]
One possible reason can be easy management of colours, with `hsl` you can do this `bg-success/50`, you will get a lighter shade of success, about 50% but with `hex` or `rgb` not possible.
